mungegithub/mungers/sig-mention-handler.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"strings"
+)
+
+type SigMentionHandler struct{}
+
+func init() {
+	h := &SigMentionHandler{}
+	RegisterMungerOrDie(h)
+}
+
+// Name is the name usable in --pr-mungers
+func (*SigMentionHandler) Name() string { return "sig-mention-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (*SigMentionHandler) RequiredFeatures() []string {
+	return []string{}
+}
+
+// Initialize will initialize the munger
+func (s *SigMentionHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (*SigMentionHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (*SigMentionHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// The algorithm:
+// (1) return false if it is a PR
+// (2) return false if the issue is closed
+// (3) return false if the issue has any label starting with "sig/"
+// (4) return true otherwise
+func (*SigMentionHandler) IsSigLabelNeeded(obj *github.MungeObject) bool {
+	if obj.Issue == nil || obj.IsPR() || obj.Issue.State == nil || *obj.Issue.State == "closed" {
+		return false
+	}
+
+	labels := obj.Issue.Labels
+
+	for i := range labels {
+		if labels[i].Name != nil && strings.HasPrefix(*labels[i].Name, "sig/") {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Munge is the workhorse notifying issue owner to add a @kubernetes/sig mention if there is none
+func (s *SigMentionHandler) Munge(obj *github.MungeObject) {
+	if !s.IsSigLabelNeeded(obj) {
```

Comment:
IMO an issue should be in one of two states:

* It has a `needs-sig` label and does not have a `sig/foo` label.
* It has a `sig/foo` label and does not have a `needs-sig` label.

This PR addresses one type of problem:
* issue has neither a `needs-sig` nor `sig/foo` label.

However it seems to me that as we start using the `needs-sig` label to find issues and add sigs we will wind up with most issues in the following invalid state:
* issue has both a `needs-sig` and `sig/foo` label.

This reduces the value of the `needs-sig` label because many issues that do not need a sig label -- they already have one -- will nonetheless still have this `needs-sig` label, making it more challenging to find issues that need a sig label.